src/main/java/de/tum/in/www1/artemis/service/plagiarism/ContinuousPlagiarismControlService.java

```diff
@@ -0,0 +1,147 @@
+package de.tum.in.www1.artemis.service.plagiarism;
+
+import static java.lang.String.format;
+
+import java.time.ZonedDateTime;
+import java.util.Set;
+import java.util.function.Predicate;
+
+import org.jvnet.hk2.annotations.Service;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.annotation.Profile;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+
+import de.jplag.exceptions.ExitException;
+import de.tum.in.www1.artemis.domain.Exercise;
+import de.tum.in.www1.artemis.domain.ProgrammingExercise;
+import de.tum.in.www1.artemis.domain.TextExercise;
+import de.tum.in.www1.artemis.domain.enumeration.DisplayPriority;
+import de.tum.in.www1.artemis.domain.metis.Post;
+import de.tum.in.www1.artemis.domain.modeling.ModelingExercise;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismCase;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismComparison;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismResult;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismStatus;
+import de.tum.in.www1.artemis.domain.plagiarism.PlagiarismSubmissionElement;
+import de.tum.in.www1.artemis.repository.ExerciseRepository;
+import de.tum.in.www1.artemis.repository.plagiarism.PlagiarismCaseRepository;
+import de.tum.in.www1.artemis.repository.plagiarism.PlagiarismComparisonRepository;
+import de.tum.in.www1.artemis.service.metis.PostService;
+import de.tum.in.www1.artemis.service.util.TimeLogUtil;
+
+/**
+ * Manages continuous plagiarism control.
+ */
+@Service
+@Component
+@Profile("scheduling")
+public class ContinuousPlagiarismControlService {
+
+    private static final Logger log = LoggerFactory.getLogger(ContinuousPlagiarismControlService.class);
+
+    private static final Predicate<Exercise> isBeforeDueDateOrAfterWithPostDueDateChecksEnabled = exercise -> exercise.getDueDate() == null
+            || exercise.getDueDate().isAfter(ZonedDateTime.now()) || exercise.getPlagiarismDetectionConfig().isContinuousPlagiarismControlPostDueDateChecksEnabled();
```

Comment:
Please use the ExerciseDateSerice. Also note that individual due date extensions are possible that should probably be taken into account here.

Reply:
We need to think about supporting individual due dates as a follow up. I'll discuss that with Stephan once we sync about the project. The `ExerciseDateService` is not necessary here at the moment, as the due date always will be non null (thanks to the database query). Nevertheless, good suggestions, thanks for noting it

Reply:
Agreed that this can be done on a later stage